dor/service_layer/handlers/verify_package.py

```diff
@@ -1,20 +1,25 @@
-
-
-from dor.domain.events import PackageReceived, PackageVerified
+from dor.adapters.bag_adapter import ValidationError
+from dor.domain.events import PackageNotVerified, PackageReceived, PackageVerified
 from dor.service_layer.unit_of_work import UnitOfWork
 
-def verify_package(event: PackageReceived, uow: UnitOfWork, bag_reader_class: type, workspace_class: type) -> None:
-    workspace = workspace_class(event.workspace_identifier)
 
-    bag_reader = bag_reader_class(workspace.package_directory())
+def verify_package(
+    event: PackageReceived, uow: UnitOfWork, bag_adapter_class: type, workspace_class: type
+) -> None:
+    workspace = workspace_class(event.workspace_identifier)
 
-    is_valid = bag_reader.is_valid()
+    bag_adapter = bag_adapter_class(workspace.package_directory())
 
-    if is_valid:
+    try:
+        bag_adapter.validate()
         uow.add_event(PackageVerified(
             package_identifier=event.package_identifier,
             tracking_identifier=event.tracking_identifier,
             workspace_identifier=workspace.identifier,
         ))
-    else:
-        raise Exception()
\ No newline at end of file
+    except ValidationError as e:
+        uow.add_event(PackageNotVerified(
+            package_identifier=event.package_identifier,
+            tracking_identifier=event.tracking_identifier,
+            message=e.message
+        ))
```

Comment:
I thought about adding an end-to-end test for this, but wasn't sure if I should do that on my own, or if we should wait until we have a handler for that event and some kind of notification component (email)?